lib/PhysicalobjectCsvImporter.class.php

```diff
@@ -0,0 +1,655 @@
+
+      case 'typeIdLookupTable':
+        return $this->getTypeIdLookupTable();
+
+        break;
+
+      default:
+        throw new sfException("Unknown or inaccessible property \"$name\"");
+    }
+  }
+
+  public function __set($name, $value)
+  {
+    switch ($name)
+    {
+      case 'dbcon':
+      case 'multiValueDelimiter':
+      case 'physicalObjectTypeTaxonomy':
+      case 'typeIdLookupTable':
+        $this->$name = $value;
+
+        break;
+
+      default:
+        throw new sfException("Couldn't set unknown property \"$name\"");
+    }
+  }
+
+  public function setFilename($filename)
+  {
+    $this->filename = $this->validateFilename($filename);
+  }
+
+  public function getFilename()
+  {
+    return $this->filename;
+  }
+
+  public function validateFilename($filename)
+  {
+    if (!file_exists($filename))
+    {
+      throw new sfException("Can not find file $filename");
+    }
+
+    if (!is_readable($filename))
+    {
+      throw new sfException("Can not read $filename");
+    }
+
+    return $filename;
+  }
+
+  public function setOptions(array $options = null)
+  {
+    if (null === $options || 0 == count($options))
+    {
+      $this->options = [];
+
+      return;
+    }
+
+    foreach ($options as $name => $val)
+    {
+      $this->setOption($name, $val);
+    }
+  }
+
+  public function getOptions()
+  {
+    return $this->options;
+  }
+
+  public function setOption(String $name, $value)
+  {
+    switch ($name)
+    {
+      case 'header':
+        $this->setHeader($value);
+
+        break;
+
+      case 'offset':
+        $this->setOffset($value);
+
+        break;
+
+      case 'updateSearchIndex':
+        $this->setUpdateSearchIndex($value);
+
+        break;
+
+      default:
+        $this->options[$name] = $value;
+    }
+  }
+
+  public function getOption(String $name)
+  {
+    if (isset($this->options[$name]))
+    {
+      return $this->options[$name];
+    }
+    elseif ('sourceName' == $name)
+    {
+      return basename($this->filename);
+    }
+  }
+
+  public function setUpdateSearchIndex($value)
+  {
+    $this->updateSearchIndex = (bool) $value;
+  }
+
+  public function getUpdateSearchIndex()
+  {
+    return $this->updateSearchIndex;
+  }
+
+  public function setOffset(int $value)
+  {
+    $this->offset = $value;
+  }
+
+  public function getOffset()
+  {
+    return $this->offset;
+  }
+
+  public function setHeader(String $str)
+  {
+    $columnNames = explode(',', trim($str));
+
+    // Trim whitespace
+    $columnNames = array_map('trim', $columnNames);
+
+    // Remove empty values
+    $columnNames = array_filter($columnNames, function ($val) {
+      return !empty($val);
+    });
+
+    if (empty($columnNames))
+    {
+      $msg = <<<EOM
+Invalid header. Please provide a CSV delimited list of column names
+e.g. "name,location,type,culture".
+EOM;
+
+      throw new sfException($msg);
+    }
+
+    // Throw error on unknown column names
+    foreach ($columnNames as $name)
```

Comment:
I think unknown columns should just be ignored and not trigger an import error. When doing migrations, I will sometimes leave additional client columns in an 'AtoM friendly' CSV for debugging/verification purposes post-import.

Reply:
@sbreker Yeah, I debated whether to throw an error on the column names.  My thinking is because these columns are passed to the importer as a task option (e.g. `--header="name,location"`) that it would be easy to typo the name of a column and none of the data in that column would import. :(   Headers included in the CSV file don't use this code.

Reply:
Ok - as long as it doesn't affect CSVs with headers included as first row then all is good!  :) Curious what the use case for --header is. I don't think this exists for any of the other tasks?

Reply:
The client who is sponsoring this feature is exporting CSV data from a barcode scanning system, and the export doesn't include column headings.